src/utils/authenticateUser.ts

```diff
@@ -0,0 +1,20 @@
+import axios from 'axios';
+
+const API_URL_PROD = 'https://api.example.com';
+
+export const authenticateUser = async (
+  username: string,
+  password: string
+): Promise<string> => {
+  const response = await axios.post(`${API_URL_PROD}/token`, {
+    username,
+    password
+  });
+
+  //check if the response is valid
+  if (response.status !== 200) {
```

Comment:
I am 90% sure that when the status is invalid, the promise rejects.
Also, this would throw an error on valid responses (2xx, 3xx).